odl/discr/discr_ops.py

```diff
@@ -258,19 +374,25 @@ def __init__(self, space, axis=0, dx=1.0, edge_order=2,
         space : `DiscreteLp`
             The space of elements which the operator is acting on
         axis : `int`, optional
-            The axis along which the partial derivative is evaluated.
-            Default: 0
+            The axis along which the partial derivative is evaluated
         dx : `float`, optional
-            Scalars specifying the sampling distances in dimension ``axis``.
-            Default distance: 1.0
-        edge_order : {1, 2}, optional
-            First-order accurate differences can be used at the boundaries
-            if no zero padding is used. Default edge order: 2
-        zero_padding : `bool`, optional
-            Implicit zero padding. Assumes values outside the domain of ``f``
-            to be zero. Default: `False`
+            Scalar specifying the distance between sampling points along
+            ``axis``. If `None` uses the side length along the ``axis`` of
+            a cell of ``space``
```

Comment:
Any reason for not using the default? The distance between sampling points is determined by the grid, so if users want a different grid, they should make a different discretization. Allowing the specification of `dx` breaks consistency.